cmd/monaco/integrationtest/v2/references_test.go

```diff
@@ -0,0 +1,245 @@
+
+				assertRefParamFromTo(t, profile, managementZone)
+				assertRefParamFromTo(t, notification, profile)
+			},
+		},
+		{
+			project: "settings",
+			downloadOpts: []string{
+				"-s", "builtin:problem.notifications,builtin:management-zones,builtin:alerting.profile",
+			},
+			validate: func(t *testing.T, ctx TestContext, confsPerType project.ConfigsPerType) {
+				managementZone := findSetting(t, confsPerType, "builtin:management-zones", "zone_"+ctx.suffix, "name")
+				profile := findSetting(t, confsPerType, "builtin:alerting.profile", "profile_"+ctx.suffix, "name")
+				notification := findSetting(t, confsPerType, "builtin:problem.notifications", "notification_"+ctx.suffix, "displayName")
+
+				assertRefParamFromTo(t, profile, managementZone)
+				assertRefParamFromTo(t, notification, profile)
+			},
+		},
+		{
+			project: "settings-with-classic-mngt-zone",
+			downloadOpts: []string{
+				"-a", "management-zone",
+				"-s", "builtin:problem.notifications,builtin:alerting.profile",
+			},
+			validate: func(t *testing.T, ctx TestContext, confsPerType project.ConfigsPerType) {
+				managementZone := findConfig(t, confsPerType, "management-zone", "zone_"+ctx.suffix)
+				profile := findSetting(t, confsPerType, "builtin:alerting.profile", "profile_"+ctx.suffix, "name")
+				notification := findSetting(t, confsPerType, "builtin:problem.notifications", "notification_"+ctx.suffix, "displayName")
+
+				assertRefParamFromTo(t, profile, managementZone)
+				assertRefParamFromTo(t, notification, profile)
+			},
+		},
+		{
+			project: "config-with-mngt-zone",
+			downloadOpts: []string{
+				"-a", "notification,alerting-profile",
+				"-s", "builtin:management-zones",
+			},
+			validate: func(t *testing.T, ctx TestContext, confsPerType project.ConfigsPerType) {
+				managementZone := findSetting(t, confsPerType, "builtin:management-zones", "zone_"+ctx.suffix, "name")
+				profile := findConfig(t, confsPerType, "alerting-profile", "profile_"+ctx.suffix)
+				notification := findConfig(t, confsPerType, "notification", "notification_"+ctx.suffix)
+
+				assertRefParamFromTo(t, profile, managementZone)
+				assertRefParamFromTo(t, notification, profile)
+			},
+		},
+	}
+
+	for _, env := range envs {
+		for _, tc := range tests {
+			testName := env + "_" + tc.project
+
+			t.Run(testName, func(t *testing.T) {
+				configFolder := "test-resources/references/"
+				manifestFile := configFolder + "manifest.yaml"
+				proj := tc.project
+
+				fs := testutils.CreateTestFileSystem()
+
+				RunIntegrationWithCleanupOnGivenFs(t, fs, configFolder, manifestFile, env, testName, func(fs afero.Fs, ctx TestContext) {
+
+					// upsert
+					cmd := runner.BuildCli(fs)
+					cmd.SetArgs([]string{"deploy", "-v", manifestFile, "--environment", env, "--project", proj})
+					err := cmd.Execute()
+					assert.Nil(t, err, "create: did not expect error")
+
+					// update just to be sure
```

Comment:
is that step really needed? If the tests fail without it, something's wrong with upserting.
If it's to ensure that the API has registered and replies with everything that was just created, why not just sleep instead?